File: packages/protocol-parser/src/encodeKeyTuple.ts
import { StaticPrimitiveType } from "@latticexyz/schema-type";
import { Hex, encodeAbiParameters } from "viem";
import { Schema } from "./common";

/** @deprecated use `encodeKey` instead */
export function encodeKeyTuple(keySchema: Schema, keyTuple: StaticPrimitiveType[]): Hex[] {
  return keyTuple.map((key, index) => encodeAbiParameters([{ type: keySchema.staticFields[index] }], [key]));
}

File: packages/services/pkg/protocol-parser/encodeKeyTuple.go
package protocolparser

import (
	schematype "latticexyz/mud/packages/services/pkg/schema-type"

	"github.com/andriidski/abiencode-go/abi"
)

func EncodeKeyTuple(keySchema Schema, keyTuple []interface{}) []string {
	encodings := []string{}
	for i, key := range keyTuple {
		encodings = append(encodings, abi.Encode(
			SchemaTypesToStringTypes([]schematype.SchemaType{keySchema.StaticFields[i]}),
			[]interface{}{key},
		))
	}
	return encodings
}
